cmd/noms/diff/diff.go

```diff
@@ -17,16 +17,27 @@ type (
 	valueFunc func(k types.Value) types.Value
 )
 
+type diffWriter struct {
+	w         io.Writer
+	leftRight bool
+}
+
+func (w diffWriter) Write(p []byte) (n int, err error) {
+	return w.w.Write(p)
+}
+
 func shouldDescend(v1, v2 types.Value) bool {
 	kind := v1.Type().Kind()
 	return !types.IsPrimitiveKind(kind) && kind == v2.Type().Kind() && kind != types.RefKind
 }
 
-func Diff(w io.Writer, v1, v2 types.Value) error {
-	return diff(w, types.Path{}, nil, v1, v2)
+// Writes the Diff from |v1| to |v2| to |w|.
```

Comment:
If you start this comment with "Diff", godoc can pull the comment out for automatic documentation I believe.

`// Diff writes the diff from....`